Mage.Sets/src/mage/cards/l/LastNightTogether.java

```diff
@@ -0,0 +1,161 @@
+    public LastNightTogether copy() {
+        return new LastNightTogether(this);
+    }
+}
+
+class LastNightTogetherEffect extends OneShotEffect {
+
+    LastNightTogetherEffect() {
+        super(Outcome.Benefit);
+        staticText = "After this main phase, there is an additional combat phase. Only the chosen creatures can attack during that combat phase";
+    }
+
+    private LastNightTogetherEffect(final LastNightTogetherEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public LastNightTogetherEffect copy() {
+        return new LastNightTogetherEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        // 15.07.2006 If it's somehow not a main phase when Fury of the Horde resolves, all it does is untap all creatures that attacked that turn. No new phases are created.
+        // Same ruling applies here
+        if (game.getTurnPhaseType() == TurnPhase.PRECOMBAT_MAIN || game.getTurnPhaseType() == TurnPhase.POSTCOMBAT_MAIN) {
+            // At the start of that combat, add a restriction effect preventing other creatures from attacking.
+            TurnMod combat = new TurnMod(game.getState().getActivePlayerId()).withExtraPhase(TurnPhase.COMBAT);
+            game.getState().getTurnMods().add(combat);
+            List<UUID> targets = source.getTargets().get(0).getTargets();
+            DelayedCantAttackAbility delayedTriggeredAbility = new DelayedCantAttackAbility(targets, game, combat.getId());
+            game.addDelayedTriggeredAbility(delayedTriggeredAbility, source);
+            return true;
+        }
+        return true;
+    }
+
+}
+
+class DelayedCantAttackAbility extends DelayedTriggeredAbility {
+
+    private UUID connectedTurnMod;
+
+    public DelayedCantAttackAbility(List<UUID> targets, Game game, UUID connectedTurnMod) {
+        super(new LastNightTogetherRestrictionEffect(targets, game), Duration.EndOfCombat);
+        this.usesStack = false; // don't show this to the user
+        this.connectedTurnMod = connectedTurnMod;
+    }
+
+    public DelayedCantAttackAbility(DelayedCantAttackAbility ability) {
+        super(ability);
+        this.connectedTurnMod = ability.connectedTurnMod;
+    }
+
+    @Override
+    public DelayedCantAttackAbility copy() {
+        return new DelayedCantAttackAbility(this);
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.PHASE_CHANGED || event.getType() == GameEvent.EventType.COMBAT_PHASE_PRE;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        return (event.getType() == GameEvent.EventType.PHASE_CHANGED && this.connectedTurnMod.equals(event.getSourceId()));
+    }
+
+    @Override
+    public String getRule() {
+        return "Only the chosen creatures can attack during that combat phase";
+    }
+}
+
+class LastNightTogetherRestrictionEffect extends RestrictionEffect {
```

Comment:
Must use standard effect like `CantAttackAllEffect` with fixed filter. See `Fight or Flight` for usage example.